### PR TITLE
planner: isolation read do not filter system tables

### LINKED ISSUE #17551: delete the `TiDB` value of `tidb_isolation_read_engines`
## Development Task

The system tables in `information_schema` and `metrics_schema` are in-memory tables of TiDB server. The queries of them are controlled by the value of `tidb_isolation_read_engines`("tidb"), which is very unfriendly for users. So we decide to delete it.

### PR DESCRIPTION
**Pull request by lzmhhh123**
### What problem does this PR solve?

Issue Number: close #17551

Problem Summary: make isolation read completely ignore system tables.

### What is changed and how it works?

What's Changed: if the type of access path is `TiDB`. Isolation read ignore it.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

Side effects

- Performance regression
    - Consumes more CPU

### Release note 

- make isolation read do not filter system tables.

### CODE REVIEW

File: planner/core/planbuilder.go [hunk 1]
```diff
@@ -826,6 +826,9 @@ func (b *PlanBuilder) filterPathByIsolationRead(paths []*util.AccessPath, dbName
 	availableEngine := map[kv.StoreType]struct{}{}
 	var availableEngineStr string
 	for i := len(paths) - 1; i >= 0; i-- {
+		if paths[i].StoreType == kv.TiDB {
```
Comment (lawyerphx) [inline planner/core/planbuilder.go:829]: will it be better to move this behind `if _, ok := availableEngine[paths[i].StoreType]; !ok {}`

**Comment by XuHuaiyu** [comment]
Does this mean that the `storeType` `TiDB` is useless?

**Comment by lzmhhh123** [comment]
> Does this mean that the `storeType` `TiDB` is useless?

Yes~

**Comment by lzmhhh123** [comment]
This is a graceful way to avoid compatibility problems. We still keep the value `tidb` and make it useless.

**Pull request merged**